miden-lib/src/transaction/procedures.rs

```diff
@@ -0,0 +1,239 @@
+use alloc::vec::Vec;
+
+use miden_objects::{Digest, Felt, Hasher};
+
+use super::TransactionKernel;
+
+macro_rules! digest {
+    ($a:expr, $b:expr, $c:expr, $d:expr) => {
+        Digest::new([Felt::new($a), Felt::new($b), Felt::new($c), Felt::new($d)])
+    };
+}
+
+impl TransactionKernel {
+    /// Hashes of all dynamically executed procedures from the kernel 0.
+    pub const KERNEL0_PROCEDURES: [Digest; 28] = [
```

Comment:
nit: I'm not sure this needs to be a public constant here - we could probably have it just as a "free" constant private to this module (e.g., we could put it into a separate section at the end of the file called `KERNEL V0 PROCEDURES`).